lib/transformer.cc

```diff
@@ -81,91 +88,130 @@ void RotateTransformer::TransformCanvas::SetPixel(int x, int y, uint8_t red, uin
   }
 }
 
-int RotateTransformer::TransformCanvas::width() const {
+void RotateTransformer::TransformCanvas::SetImage(uint8_t *rawdata, int length, bool isBGR)
```

Comment:
This implementation should go into the base Canvas class: it is always the same for everyone.
It can still be virtual, so that  special implementations might be able to override it if they have a faster implementation for their use-case.